docs/reference/pages/model_recipes.md

```diff
@@ -0,0 +1,277 @@
+(page_model_recipes)=
+
+# Recipes
+
+## Overriding the {meth}`~wagtail.models.Page.serve` Method
+
+Wagtail defaults to serving {class}`~wagtail.models.Page`-derived models by passing a reference to the page object to a Django HTML template matching the model's name, but suppose you wanted to serve something other than HTML? You can override the {meth}`~wagtail.models.Page.serve` method provided by the {class}`~wagtail.models.Page` class and handle the Django request and response more directly.
+
+Consider this example of an `EventPage` object which is served as an iCal file if the `format` variable is set in the request:
+
+```python
+class EventPage(Page):
+    ...
+
+    def serve(self, request):
+        if "format" in request.GET:
+            if request.GET['format'] == 'ical':
+                # Export to ical format
+                response = HttpResponse(
+                    export_event(self, 'ical'),
+                    content_type='text/calendar',
+                )
+                response['Content-Disposition'] = 'attachment; filename=' + self.slug + '.ics'
+                return response
+            else:
+                # Unrecognised format error
+                message = 'Could not export event\n\nUnrecognised format: ' + request.GET['format']
+                return HttpResponse(message, content_type='text/plain')
+        else:
+            # Display event page as usual
+            return super().serve(request)
+```
+
+{meth}`~wagtail.models.Page.serve` takes a Django request object and returns a Django response object. Wagtail returns a `TemplateResponse` object with the template and context which it generates, which allows middleware to function as intended, so keep in mind that a simpler response object like a `HttpResponse` will not receive these benefits.
+
+With this strategy, you could use Django or Python utilities to render your model in JSON or XML or any other format you'd like.
+
+(overriding_route_method)=
+
+### Adding Endpoints with Custom {meth}`~wagtail.models.Page.route` Methods
```

Comment:
Nice, this does make more sense to me as a sub-heading. And even more so for the tagging ones below.